### PR TITLE
fix: [CI-9213]: Added code for  building JsonObject from ambiance for release event

### PR DESCRIPTION
**Pull request by sahithibanda01**
## Describe your changes
For event type "release" code base variables were not getting populated. Added the missing change for that
Tested for the codebase variables , they are now populating
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51852)

### CODE REVIEW

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/helpers/TriggerHelper.java [hunk 1]
```diff
@@ -118,6 +90,15 @@ public Map<String, Object> buildJsonObjectFromAmbiance(TriggerPayload triggerPay
           jsonObject.put(TAG, parsedPayload.getPush().getRef().replaceFirst("refs/tags/", ""));
         }
         break;
+      case RELEASE:
+        jsonObject.put(BRANCH, parsedPayload.getRelease().getRepo().getBranch());
+        jsonObject.put(TARGET_BRANCH, parsedPayload.getRelease().getRepo().getBranch());
+        jsonObject.put(EVENT, RELEASE_EVENT_TYPE);
+        jsonObject.put(TYPE, WEBHOOK_TYPE);
+        jsonObject.put(REPO_URL, parsedPayload.getRelease().getRepo().getLink());
+        jsonObject.put(GIT_USER, parsedPayload.getRelease().getRepo().getNamespace());
```
Comment (raghavharness) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/helpers/TriggerHelper.java:99]: Should be `parsedPayload.getRelease().getSender().getLogin()`

File: 990-commons-test/src/main/java/io/harness/rule/OwnerRule.java [hunk 2]
```diff
@@ -579,6 +580,7 @@ private static UserInfoBuilder defaultUserInfo(String user) {
           .put(MTRAN, defaultUserInfo(MTRAN).slack("UCYBPAE3Y").team(PLG).build())
           .put(SHASHWAT_SACHAN, defaultUserInfo(SHASHWAT_SACHAN).slack("U05GP44JNJ2").team(CV).build())
           .put(AYUSHI_TIWARI, defaultUserInfo(AYUSHI_TIWARI).slack("U05KLP7V9J5").team(SPG).build())
+          .put(SAHITHI, defaultUserInfo(SAHITHI).email("sahithi.banda@harness.io").team(CDC).build())
```
Comment (raghavharness) [inline 990-commons-test/src/main/java/io/harness/rule/OwnerRule.java:583]: Can you add you slack ID here

**Review by raghavharness — APPROVED** [review]
LGTM

**Comment by sahithibanda01** [comment]
trigger uts

**Review by archit-harness — APPROVED** [review]

**Pull request merged**